pkg/provider/azure_instances.go

```diff
@@ -231,6 +231,10 @@ func (az *Cloud) InstanceExists(ctx context.Context, node *v1.Node) (bool, error
 		var err error
 		providerID, err = cloudprovider.GetInstanceProviderID(ctx, az, types.NodeName(node.Name))
 		if err != nil {
+			if strings.Contains(err.Error(), cloudprovider.InstanceNotFound.Error()) {
```

Comment:
why can't we use `error.Is` here?

Reply:
also when will `GetInstanceProviderId` return `InstanceNotFound`? I don't see that happening in any codepath.

https://github.com/kubernetes/cloud-provider/blob/master/cloud.go#L101-L116

Reply:
this is because the error would be wrapped into `fmt.Errorf("failed to get instance ID from cloud provider: %v", err)` inside `cloudprovider.GetInstanceProviderID`.

Reply:
> why can't we use error.Is here?

https://github.com/kubernetes/cloud-provider/blob/master/cloud.go#L112 use `%v` instead of `%w` which I think [errors.Unwrap](https://pkg.go.dev/errors#Unwrap) could not work.

> also when will GetInstanceProviderId return InstanceNotFound? I don't see that happening in any codepath.

https://github.com/kubernetes/cloud-provider/blob/master/cloud.go#L174-L177 the interface requires the impl should return cloudprovider.InstanceNotFound.
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/pkg/provider/azure_instances.go#L376 plz check this impl, it would return `InstanceNotFound`.